src/Monolog/Handler/StreamHandler.php

```diff
@@ -97,8 +108,32 @@ protected function write(array $record)
         }
     }
 
-    private function customErrorHandler($code, $msg)
+    private function customMkDirErrorHandler($code, $msg)
+    {
+        $this->errorMessage = $msg;
+    }
+
+    private function customFopenErrorHandler($code, $msg)
     {
         $this->errorMessage = preg_replace('{^fopen\(.*?\): }', '', $msg);
```

Comment:
I think you could use the same customErrorHandler for both, just change fopen here to be `(fopen|mkdir)`, the regex only cleans up php's error message which is like: `foo(arg): ACTUAL MESSAGE`.